apps/sms/services/test/unit/messaging/messaging_service_test.js

```diff
@@ -0,0 +1,628 @@
+/*global bridge,
+         MocksHelper,
+         MockMessages,
+         Settings,
+         SMIL
+*/
+
+'use strict';
+
+require('/services/js/bridge_service_mixin.js');
+require('/services/test/unit/mock_bridge.js');
+require('/views/shared/test/unit/mock_messages.js');
+require('/views/shared/test/unit/mock_settings.js');
+require('/views/shared/test/unit/mock_smil.js');
+
+var MocksHelperForAttachment = new MocksHelper([
+  'bridge',
+  'Settings',
+  'SMIL'
+]).init();
+
+suite('Messaging service >', function() {
+  var MessagingService, successResult, errorResult;
+
+  MocksHelperForAttachment.attachTestHelpers();
+
+  suiteSetup(function(done) {
+    window.importScripts = sinon.stub();
+
+    require('/services/js/messaging/messaging_service.js', done);
+  });
+
+  suiteTeardown(function() {
+    delete window.importScripts;
+  });
+
+  setup(function() {
+    MessagingService = self.MessagingService;
```

Comment:
nit: ```self``` is global context aka window, so need to redefine, we can just reference ```MessagingService``` directly.

Also can't we just leverage this setup to stub both client and service in the same way and don't abuse private ```MessagingService.mozMobileMessageClient```, like this (basically we'll just have ```'service mixing and client are initialized'``` test instead of ```init``` suite)?:
```
setup(function() {
  successResult = {};
  errorResult = new Error();

  serviceStub = sinon.stub({
    method: () => {},
    stream: () => {},
    broadcast: () => {},
    listen: () => {},
    plugin: () => {}
  });

  mozMobileMessageClientStub =  {
    method: sinon.stub()
  };

  this.sinon.stub(bridge, 'service').returns(serviceStub);
  this.sinon.stub(bridge, 'client').returns(mozMobileMessageClientStub);

  MessagingService.init();
});
```